Use unittest assertWarns instead

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31860 [jit] Add dictionary as output of tracer
* #36277 [jit] tracer flag to guard risky behaivors
* **#36411 Use unittest assertWarns instead**

This PR remove pytorch specific defined assertwarns and use the unit
test one, also format some tests

Differential Revision: [D20998159](https://our.internmc.facebook.com/intern/diff/D20998159)

wanchaol committed:
Use unittest assertWarns instead

This PR remove pytorch specific defined assertwarns and use the unit
test one, also format some tests

[ghstack-poisoned]

mrshenli commented on test/test_jit.py:
```diff
@@ -406,7 +406,7 @@ class M(torch.jit.ScriptModule):
                 super(M, self).__init__()
                 self.p0 = nn.Parameter(torch.tensor([0.3], dtype=torch.float,
-                                       device=cpu_device_str))
+                                                    device=cpu_device_str))
```
indent?

mrshenli commented on test/distributed/test_data_parallel.py:
```diff
@@ -645,8 +645,9 @@ def __init__(self, testcase):
 
             def forward(self, x):
                 self._testcase.assertWarnsRegex(
```
should this also use `with` statement?

mrshenli reviewed (approved):
stamping for data_parallel related changes.

wanchaol committed:
Update on "Use unittest assertWarns instead"


This PR remove pytorch specific defined assertwarns and use the unit
test one, also format some tests

[ghstack-poisoned]

wanchaol committed:
Update on "Use unittest assertWarns instead"


This PR remove pytorch specific defined assertwarns and use the unit
test one, also format some tests

[ghstack-poisoned]
